integration/update_test.go

```diff
@@ -72,3 +73,186 @@ func TestUpdateUpsert(t *testing.T) {
 	require.NoError(t, err)
 	AssertEqualDocuments(t, bson.D{{"_id", id}, {"foo", "qux"}}, doc)
 }
+
+func TestUpdateIncOperatorErrors(t *testing.T) {
+	t.Parallel()
+
+	for name, tc := range map[string]struct {
+		filter bson.D
+		update bson.D
+		err    *mongo.WriteError
```

Comment:
why not create bare value, not a pointer? 
Mongo driver returns not a pointer but a bare struct.

Reply:
ah I see now - to make nil pointer check